src/Filters/LinearKalmanFilter.cpp

```diff
@@ -0,0 +1,63 @@
+#include "LinearKalmanFilter.h"
+
+namespace mmfs
+{
+    LinearKalmanFilter::LinearKalmanFilter(Matrix X, Matrix U, Matrix P, Matrix F, Matrix G, Matrix R)
+    {
+        state.X = X;
+        state.U = U;
+        state.P = P;
+        state.F = F;
+        state.G = G;
+        state.R = R;
+        calculate_initial_values();
+    }
+
+    void LinearKalmanFilter::predict_state()
+    {
+        state.X = (state.F * state.X) + (state.G * state.U);
+    }
+
+    void LinearKalmanFilter::estimate_state(Matrix measurement)
+    {
+        state.X = state.X + state.K * (measurement - state.H * state.X);
+    }
+
+    void LinearKalmanFilter::calculate_kalman_gain()
+    {
+        state.K = state.P * state.H.T() * (state.H * state.P * state.H.T() + state.R).inv();
+    }
+
+    void LinearKalmanFilter::covariance_update()
+    {
+        int n = state.X.getRows();
+        state.P = ((Matrix::ident(n) - state.K * state.H) * state.P * ((Matrix::ident(n) - state.K * state.H).T())) + state.K * state.R * state.K.T();
+    }
+
+    void LinearKalmanFilter::covariance_extrapolate()
+    {
+        state.P = state.F * state.P * state.F.T() + state.Q;
+    }
+
+    void LinearKalmanFilter::calculate_initial_values()
+    {
+        state.Q = (state.G * 0.2 * 0.2) * state.G.T();
```

Comment:
what are these .2?? It they are a hardcoded number, make them a variable that is defined somewhere easy to find